From 610c00a51c994d8f7eb9c333002b0ba8ada0ae7e Mon Sep 17 00:00:00 2001
From: Matthew Gerlach <mgerlach@opensource.altera.com>
Date: Tue, 3 May 2016 08:48:27 -0700
Subject: [PATCH 141/142] FogBugz #374571: Fix Sysid driver hotplug support.

The Altera Sysid component is generally part of
of an FPGA design.  The component can be hotplugged
when the FPGA is reconfigured.  This patch fixes the
driver to support the component being hotplugged.

Signed-off-by: Matthew Gerlach <mgerlach@opensource.altera.com>
[Original patch taken from
 https://github.com/altera-opensource/linux-socfpga.git
 tag rel_socfpga-4.8_17.03.01_pr]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 drivers/misc/altera_sysid.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/misc/altera_sysid.c b/drivers/misc/altera_sysid.c
index ff2b2b5bb0a3..be35530c504b 100644
--- a/drivers/misc/altera_sysid.c
+++ b/drivers/misc/altera_sysid.c
@@ -118,13 +118,13 @@ static struct platform_driver altera_sysid_platform_driver = {
 		.owner		= THIS_MODULE,
 		.of_match_table	= of_match_ptr(altera_sysid_match),
 	},
+	.probe			= altera_sysid_probe,
 	.remove			= altera_sysid_remove,
 };
 
 static int __init altera_sysid_init(void)
 {
-	return platform_driver_probe(&altera_sysid_platform_driver,
-		altera_sysid_probe);
+	return platform_driver_register(&altera_sysid_platform_driver);
 }
 
 static void __exit altera_sysid_exit(void)
-- 
2.11.0

